share/scripts/tcl_bridge.tcl

```diff
@@ -0,0 +1,117 @@
+# tcl_bridge v1.0 (c) 2025 Pedro de Medeiros
+# There is an example program that consumes this script at:
+# https://github.com/pvmm/openmsx-tcl-bridge
+#
+# TODO: add connection structure description here
+#
+namespace eval tcl_bridge {
+
+variable wp {}
+variable latch 0
+variable address
+variable echo 0
+
+set_help_text tcl_bridge \
+{tcl_bridge -- allows guest/host communication through a MSX-to-Tcl bridge.
+
+Usage:
+  tcl_bridge start      Create communication channel
+  tcl_bridge stop       Destroy communication channel
+  tcl_bridge echo {1|0} Turn on/off echoing command result to console
+}
+
+proc tcl_bridge {args} {
+    set cmd [lindex $args 0]
+    set param [lindex $args 1]
+    switch -- $cmd {
+        start   { tcl_bridge::start }
+        stop    { tcl_bridge::stop }
+        echo    { return [tcl_bridge::echo $param] }
+        default { error "Unknown sub-command \"$cmd\"." }
+    }
+}
+
+proc start {} {
+    variable wp
+    if {$wp eq {}} {
+        set wp [debug watchpoint create -type write_io -address 6 -command {tcl_bridge::read_command $::wp_last_value}]
+    }
+}
+
+proc stop {} {
+    variable wp
+    if {$wp ne {}} {
+        debug watchpoint remove $wp
+        set wp {}
+    }
+}
+
+proc echo {status} {
+    variable echo
+    if {$status ne {}} { set echo $status }
+    return $echo
+}
+
+proc min {n1 n2} {
+    if {$n2 > $n1} { return $n1 }
+    return $n2
+}
+
+proc execute_command {base} {
+    set input_addr  [peek16 $base]                      ;# input address
+    set input_size  [peek16 [expr {$base + 2}]]         ;# input size value
+    set output_addr [peek16 [expr {$base + 4}]]         ;# output address
+    set output_size_addr    [expr {$base + 6}]          ;# output size address
+    set output_max  [peek16 [expr {$base + 8}]]         ;# max output buffer size
+    set status_addr         [expr {$base + 10}]         ;# status address
+    set c {}
+    set s ""
+    if {[expr {$input_addr + $input_size}] < 0xFFFF} {
+        for {set i 0} {$i < $input_size} {incr i} {
+            set c [peek [expr {$input_addr + $i}]]
+            append s [format %c $c]
+        }
+    }
+    if {$c ne {}} {
+        # execute command
+        if {[catch {set output [eval $s]} output]} {
+            set status 1
+        } else {
+            set status 0
+        }
+    } else {
+        # overflow
+        set status 2
+        set output {}
```

Comment:
As I (tried to) explain above, we shouldn't manually check for overflow (because the `debug read` or `peek`) command already does this. So this check for `$c ne {}` and this `else` block isn't needed.